tests/unit/data-table/DataTable.spec.js

```diff
@@ -0,0 +1,204 @@
+        })
+
+        it('Sorts a mix of strings and numbers naturally ascending', () => {
+            const localThis = {
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    { value: '1 dog' },
+                    { value: 12 },
+                    { value: '21 dogs' },
+                    { value: '3 dogs' },
+                    { value: 30 }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'asc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual(['1 dog', '3 dogs', 12, '21 dogs', 30])
+        })
+
+        it('Sorts a mix of strings and numbers naturally descending', () => {
+            const localThis = {
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    { value: 'device 1' },
+                    { value: 'device 3' },
+                    { value: 'device 21' }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'desc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual(['device 21', 'device 3', 'device 1'])
+        })
+
+        it('Treats null and undefined as equivalent to empty string', () => {
+            const localThis = {
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    {},
+                    { value: 'zebra' },
+                    { value: null },
+                    { value: 'apple' },
+                    { value: '' }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'asc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual([undefined, null, '', 'apple', 'zebra'])
+        })
+
+        it('Sorts booleans descending', () => {
+            const localThis = {
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    { value: true },
+                    { value: false },
+                    { value: true }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'desc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual([false, true, true])
+        })
```

Comment:
Should descending not return `[true, true, false]`?

Reply:
At the time, I thought not, but I agree that one could argue that true is larger than false, changed in 1bbf300b66cbf6bc4d617de0ee94dc53594dd670